### PR TITLE
Add Windows Fallback, make capnpc a feature

### PR DESCRIPTION
**Pull request by ErikMcClure**
If developer mode is not turned on, cxxbuild cannot make a symlink and will instead copy over only header files, and will deliberately *not* copy over header files from a sub-project, which makes it impossible to align with capnpc's nested folder structure. This adds code that copies over all source files, and also removed capnpc as a separate project, instead making it a default feature of capnp. This should work on any version of windows and any version of linux.

### CODE REVIEW

File: c++/src/kj/build.rs [hunk 1]
```diff
@@ -27,6 +29,7 @@ static KJ_SOURCES_LITE: &[&str] = &[
     "test-helpers.c++",
     "thread.c++",
     "units.c++",
+    "time.c++", // capnproto incorrectly relies on this in mutex.c++ on windows only.
```
Comment (LunNova) [inline c++/src/kj/build.rs:32]: worth only including when CARGO_BUILD_TARGET is win?

File: c++/src/capnp/build.rs [hunk 2]
```diff
@@ -95,36 +111,41 @@ fn main() -> eyre::Result<()> {
...
-        .try_for_each(|p| {
-            println!(
-                "cargo:rerun-if-changed={}",
-                p.to_str().ok_or_eyre("non–UTF-8 path")?
-            );
+        .chain(if CAPNPC { CAPNPC_SOURCES } else { &[] })
+        .map(|s| (s, source_dir.join(s)))
+        .for_each(|(s, p)| {
+            println!("cargo:rerun-if-changed={}", s);
+            // This copy is only here in case the symlink fails on windows
```
Comment (LunNova) [inline c++/src/capnp/build.rs:118]: does this risk leaving stale files?  
windows with symlinks off and copy: gets copied once, never again if source updates. unsure if dir this is copying to is one that'd be wiped out in any case where the source changes so it doesn't matter

Comment (LunNova) [reply c++/src/capnp/build.rs:118]: iirc hard links work on windows without special configg

Comment (ErikMcClure) [reply c++/src/capnp/build.rs:118]: cxxbuild tries to delete the folder before making it, so assuming this doesn't also break for some reason, it should never risk leaving stale files: https://github.com/dtolnay/cxx/blob/a7898a27add721eca22184acc2d22dd7022c47c4/gen/build/src/out.rs#L70

**Review by LunNova — APPROVED** [review]

**Pull request merged**